### PR TITLE
Fix appending to existing tables with truncated column names

### PR DESCRIPTION
**Pull request by crisptrutski**
### Description

This fixes an issue where uploads created using a previous version of Metabase could not be appended to, even with newer versions.

See https://github.com/metabase/metabase/pull/44727#issuecomment-2192717280 for a good explanation of this bug.

If we had already released the related PR, we'd have been unable to support appends to both older and newer tables, so it's lucky that we caught this quickly.

### CODE REVIEW

File: src/metabase/upload/util.cljc [hunk 1]
```diff
@@ -0,0 +1,58 @@
+(ns metabase.upload.util)
+
+(defn unique-name-generator
+  "A fork of [[mbql.u/unique-name-generator]] which accounts for collisions when names are truncated.
+  There is due to a limitation where it does not play well with truncation, see the tests for this function.
```
Comment (crisptrutski) [inline src/metabase/upload/util.cljc:5]: I should reword this, but the limitation is with the `mbql.u` version, not this one.

File: src/metabase/upload/util.cljc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  This may be a bit slower, and it's a breaking change as far as what ids look like, hence it's a fork."
```
Comment (crisptrutski) [inline src/metabase/upload/util.cljc:6]: The breaking change is that we would no longer produce IDs like `blah_2_2`, rather it would become `blah_3`.

**Comment by calherries** [comment]
See my comment [here](https://github.com/metabase/metabase/issues/44725#issuecomment-2195780743). tl;dr I strongly feel this solution is introducing too much magic.

**Comment by crisptrutski** [comment]
Going to rework this with a simpler version that uses an MBQL convention of using hashes for really long names.

**Pull request closed without merging**